src/utils/config.py

```diff
@@ -0,0 +1,252 @@
+"""Simple utils to assist the config."""
+import pathlib
+import os
+from copy import deepcopy
+from typing import Union
+
+import yaml
+
+from src.utils.defence import (
+    _type_defence, 
+    _validate_file_extension
+)
+
+def safeload_yaml(path: Union[str, pathlib.Path]) -> dict:
+    """Load a .yaml file from a path.
+
+    Args:
+        path (Union[str, pathlib.Path]): The path to load the .yaml file from.
+
+    Raises:
+        FileNotFoundError: Raised if there is no file at the given path.
+        TypeError: Raised if the file does not have the .yaml extension.
+
+    Returns:
+        dict: The loaded yaml file as as dictionary.
+    """
+    if not os.path.exists(path):
```

Comment:
Done. Test not built for HDFS function at this time. This will require more thinking (writing to a tmp path in hdfs? not 100% sure how this would work as of now)